chatkit-core/src/main/kotlin/com/pusher/chatkit/users/api/UserSubscriptionDispatcher.kt

```diff
@@ -7,16 +7,21 @@ import com.pusher.chatkit.state.JoinedRoomsReceived
 import com.pusher.chatkit.state.LeftRoom
 import com.pusher.chatkit.state.RoomDeleted
 import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.subscription.SubscriptionListener
+import elements.EOSEvent
+import elements.Error
+import elements.Headers
+import elements.SubscriptionEvent
 import org.reduxkotlin.Dispatcher
 
 internal class UserSubscriptionDispatcher(
     private val joinedRoomsStateDiffer: JoinedRoomsStateDiffer,
     private val joinedRoomApiTypeMapper: JoinedRoomApiTypeMapper,
     private val dispatcher: Dispatcher
-) {
+) : SubscriptionListener<UserSubscriptionEvent> {
 
-    internal fun onEvent(event: UserSubscriptionEvent) {
-        when (event) {
+    override fun onEvent(elementsEvent: SubscriptionEvent<UserSubscriptionEvent>) {
+        when (val event = elementsEvent.body) {
```

Comment:
it's not nice to have to consume it like that in our dispatchers, can we alter SubscriptionListener and do that unpackaging in the platform listeners created within `asPlatformListeners`?

Reply:
SubscriptionEvent is a platform's raw type and I don't expect we'll ever need anything else that the parsed body, indeed we do not need it in v1

Reply:
We don't need the envelope right now, but we will in future (to version the state using event IDs for resuming message subscriptions when we have persistence of the state).

Am I thinking too far ahead in that?

Reply:
oh ok, sorry forgot about it, then you're right and thinking indeed very well!

Reply:
if we keep that I think it's fine to keep the rest of "raw" bits so that everything stays at the same level of abstraction, sorry for not the best feedback!